engine/apps/slack/scenarios/distribute_alerts.py

```diff
@@ -50,69 +50,147 @@
             ).update(slack_message_sent=True, reason_to_skip_escalation=AlertGroup.RATE_LIMITED)
             return
 
-        num_updated_rows = AlertGroup.objects.filter(pk=alert.group.pk, slack_message_sent=False).update(
+        num_updated_rows = AlertGroup.objects.filter(pk=alert_group_pk, slack_message_sent=False).update(
             slack_message_sent=True
         )
 
         if num_updated_rows == 1:
+            # this will be the case in the event that we haven't yet created a Slack message for this alert group
+
+            slack_channel = (
+                alert_group.channel_filter.slack_channel
+                if alert_group.channel_filter
+                # if channel filter is deleted mid escalation, use default Slack channel
+                else alert_receive_channel.organization.default_slack_channel
+            )
+            slack_channel_id = slack_channel.slack_id
+
             try:
-                slack_channel = (
-                    alert.group.channel_filter.slack_channel
-                    if alert.group.channel_filter
-                    # if channel filter is deleted mid escalation, use default Slack channel
-                    else alert.group.channel.organization.default_slack_channel
+                self._post_alert_group_to_slack(
+                    slack_team_identity=slack_team_identity,
+                    alert_group=alert_group,
+                    alert=alert,
+                    attachments=alert_group.render_slack_attachments(),
+                    slack_channel=slack_channel,
+                    blocks=alert_group.render_slack_blocks(),
                 )
-                self._send_first_alert(alert, slack_channel)
             except (SlackAPIError, TimeoutError):
-                AlertGroup.objects.filter(pk=alert.group.pk).update(slack_message_sent=False)
+                AlertGroup.objects.filter(pk=alert_group_pk).update(slack_message_sent=False)
                 raise
 
-            if alert.group.channel.maintenance_mode == AlertReceiveChannel.DEBUG_MAINTENANCE:
-                self._send_debug_mode_notice(alert.group, slack_channel)
+            if alert_receive_channel.maintenance_mode == AlertReceiveChannel.DEBUG_MAINTENANCE:
+                # send debug mode notice
+                text = "Escalations are silenced due to Debug mode"
+                self._slack_client.chat_postMessage(
+                    channel=slack_channel_id,
+                    text=text,
+                    attachments=[],
+                    thread_ts=alert_group.slack_message.slack_id,
+                    mrkdwn=True,
+                    blocks=[
+                        {
+                            "type": "section",
+                            "text": {
+                                "type": "mrkdwn",
+                                "text": text,
+                            },
+                        },
+                    ],
+                )
```

Comment:
same thing here, moved `_send_debug_mode_notice` inline here because it's only called once (here) and is simply making a call to `self._slack_client.chat_postMessage`